__tests__/Unit/Components/Tasks/TaskDependency.test.tsx

```diff
@@ -0,0 +1,130 @@
+            },
+        },
+        {
+            status: 'fulfilled',
+            value: {
+                title: 'Dependency 2',
+                id: 'dependency-2',
+            },
+        },
+    ];
+
+    test('renders loading state', () => {
+        render(
+            <TaskDependency
+                loading={true}
+                fetching={false}
+                error={false}
+                dependencyData={[]}
+                navigateToTask={() => Object}
+                isEditing={true}
+                updatedDependencies={[]}
+                handleChange={() => Object}
+            />
+        );
+
+        expect(screen.getByText(/Loading/i)).toBeInTheDocument();
+    });
+
+    test('renders error state', () => {
+        render(
+            <TaskDependency
+                loading={false}
+                fetching={false}
+                error={true}
+                dependencyData={[]}
+                navigateToTask={() => Object}
+                isEditing={true}
+                updatedDependencies={[]}
+                handleChange={() => Object}
+            />
+        );
+
+        expect(
+            screen.getByText(/Unable to fetch dependency tasks/i)
+        ).toBeInTheDocument();
+    });
+
+    test('renders no dependencies message', () => {
+        render(
+            <TaskDependency
+                loading={false}
+                fetching={false}
+                error={false}
+                dependencyData={[]}
+                navigateToTask={() => Object}
+                isEditing={true}
+                updatedDependencies={[]}
+                handleChange={() => Object}
+            />
+        );
+
+        expect(screen.getByText(/No dependencies/i)).toBeInTheDocument();
+    });
+
+    test('renders dependency list', () => {
+        render(
+            <TaskDependency
+                loading={false}
+                fetching={false}
+                error={false}
+                dependencyData={dependencyDataMock}
+                navigateToTask={() => Object}
+                isEditing={true}
+                updatedDependencies={[]}
+                handleChange={() => Object}
+            />
+        );
+
+        const dependencyItems = screen.getAllByRole('listitem');
+        expect(dependencyItems).toHaveLength(dependencyDataMock.length);
+
+        dependencyDataMock.forEach((dependency) => {
+            const dependencyTitle = screen.getByText(dependency.value.title);
+            expect(dependencyTitle).toBeInTheDocument();
+        });
+    });
+    it('should update editedDependencies state and call handleChange when dependencies change', () => {
+        const handleChange = jest.fn();
+
+        const { getByRole } = render(
+            <TaskDependency
+                loading={false}
+                fetching={false}
+                error={false}
+                dependencyData={[]}
+                navigateToTask={() => Object}
+                isEditing={true}
```

Comment:
Is there a special case for when `isEditing` is false? If so please test it

Reply:
remove